src/settings.mli

```diff
@@ -36,6 +36,7 @@ val substitute_workspace_vars : string -> string
 val server_extraEnv : unit -> string Interop.Dict.t option
 val server_args_setting : string list setting
 val server_codelens_setting : bool setting
+val server_codelens_for_nested_bindings_setting : bool option setting
```

Comment:
It feels wrong to have a `bool option setting`, I don't think vscode has UI for that.

Let's stick with a non-nullable bool and default to `false` when decoding if the field is absent using Jsonoo's `try_default` instead of `try_optional`). WDYT ?